packages/x-charts/src/BarChart/BarPlot.tsx

```diff
@@ -45,11 +48,11 @@ function getBandSize({
   };
 }
 
-export interface BarPlotSlots extends BarElementSlots {}
+export interface BarPlotSlots extends BarElementSlots, BarLabelSlots {}
 
-export interface BarPlotSlotProps extends BarElementSlotProps {}
+export interface BarPlotSlotProps extends BarElementSlotProps, BarLabelSlotProps {}
 
-export interface BarPlotProps extends Pick<BarElementProps, 'slots' | 'slotProps'> {
+export interface BarPlotProps {
```

Comment:
Why removing this?

Reply:
We need to also add the new `BarLabelSlots`. And typescript complains if we do 

```tsx
export interface BarPlotProps
  extends Pick<BarElementProps, 'slots' | 'slotProps'>,
    Pick<BarLabelProps, 'slots' | 'slotProps'> {}
```

```tsx
Interface 'BarPlotProps' cannot simultaneously extend types 
'Pick<BarElementProps, "slots" | "slotProps">' 
and 'Pick<BarLabelProps, "slots" | "slotProps">'.
```

Also it seemed simpler to add them directly based on the above extended props.

Reply:
Make sens

On this same part of the code, you could use `Pick<BarLabelProps, 'barLabel'>` to avoididuplicating the type of the `barLabel` which is just passed From `BarPlot` to `BarLabelPlot`